engine/src/main/scala/cromwell/services/EngineMetadataServiceActor.scala

```diff
@@ -83,6 +83,14 @@ case class EngineMetadataServiceActor(serviceConfig: Config, globalConfig: Confi
     }
   }
 
+  private def queryWorkflowOutputsAndRespond(id: WorkflowId): Unit = {
+    val replyTo = sender()
+    dataAccess.queryWorkflowOutputs(id) onComplete {
+      case Success(o) => replyTo ! WorkflowOutputsResponse(id, o)
+      case Failure(t) => replyTo ! WorkflowOutputsFailure(id, t)
```

Comment:
Is there logic anywhere which can tell the difference between no-such-id and no-outputs-for-this-workflow?